### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This is not a full run of check-spelling. It's _mostly_ limited to `.qhelp` files -- The action's sarif support is very new; codeql doesn't like >5000 item reports, and I hadn't written the code to automatically trim (I've drafted some thoughts but won't be able to implement them this week).

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/codeql/actions/runs/3214090382#summary-8787786982

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/codeql/commit/30e5caf5d61dda5a8d180f829ba9b83097056234
 
Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: cpp/ql/src/Best Practices/Magic Constants/MagicConstantsNumbers.qhelp [hunk 1]
```diff
@@ -20,7 +20,7 @@ update if the requirements change (for example, one more guest is permitted).
 </li>
 </ol>
 
-<p>This rule finds magic numbers for which there is no pre-existing named
+<p>This rule finds magic numbers for which there is no preexisting named
```
Comment (jsoref) [inline cpp/ql/src/Best Practices/Magic Constants/MagicConstantsNumbers.qhelp:23]: I'm happy to drop any particular change(s). This is a word, and thus there's no particular reason to use the hyphenated flavor.

File: cpp/ql/src/Best Practices/Unused Entities/UnusedStaticVariables.qhelp [hunk 2]
```diff
@@ -11,7 +11,7 @@ caused by an unhandled case.</p>
 
 </overview>
 <recommendation>
-<p>Check that the unused static variable does not indicate a defect, for example, an unhandled case. If the static variable is genuinuely not needed, 
+<p>Check that the unused static variable does not indicate a defect, for example, an unhandled case. If the static variable is genuinely not needed,
```
Comment (jsoref) [inline cpp/ql/src/Best Practices/Unused Entities/UnusedStaticVariables.qhelp:14]: I'm assuming this (and similar items) is (are) user facing.

File: cpp/ql/src/Critical/DescriptorMayNotBeClosed.qhelp [hunk 3]
```diff
@@ -19,7 +19,7 @@ This can occur when an operation performed on the open descriptor fails, and the
 
 <example>
 <p>In the example below, the <code>sockfd</code> socket may remain open if an error is triggered. 
-The code should be updated to ensure that the socket is always closed when when the function ends.
+The code should be updated to ensure that the socket is always closed when the function ends.
```
Comment (jsoref) [inline cpp/ql/src/Critical/DescriptorMayNotBeClosed.qhelp:22]: Sometimes there's a reason for doubled words. There didn't seem to be one here.

File: cpp/ql/src/Likely Bugs/Likely Typos/IncorrectNotOperatorUsage.qhelp [hunk 4]
```diff
@@ -6,9 +6,9 @@
...
+<p>Due to the nature of logical operation result value, only the lowest bit could possibly be set, and it is unlikely to be intent in bitwise operations. Violations are often indicative of a typo, using a logical-not (<code>!</code>) operator instead of the bit-wise not (<code>~</code>) operator. </p>
 <p>This rule is restricted to analyze bit-wise and (<code>&amp;</code>) and bit-wise or (<code>|</code>) operation in order to provide better precision.</p>
-<p>This rule ignores instances where a double negation (<code>!!</code>) is explicitly used as the opeartor of the bitwise operation, as this is a commonly used as a mechanism to normalize an integer value to either 1 or 0.</p>
+<p>This rule ignores instances where a double negation (<code>!!</code>) is explicitly used as the operator of the bitwise operation, as this is a commonly used as a mechanism to normalize an integer value to either 1 or 0.</p>
 <p>NOTE: It is not recommended to use this rule in kernel code or older C code as it will likely find several false positive instances.</p>
```
Comment (jsoref) [inline cpp/ql/src/Likely Bugs/Likely Typos/IncorrectNotOperatorUsage.qhelp:12]: This gave me a headache, since the file is in a directory named `Likely Typos` and thus my initial response was "oops, I better ignore this file". But, this doesn't appear to be the likely typos it was talking about.

File: csharp/ql/src/Bad Practices/Implementation Hiding/AbstractToConcreteCollection.qhelp [hunk 5]
```diff
@@ -18,7 +18,7 @@ more difficult to change which implementation you are using at a later date.</p>
 </example>
 <references>
 
-  <li>C# Corner, <a href="http://www.c-sharpcorner.com/UploadFile/rmcochran/csharp_interrfaces03052006095933AM/csharp_interrfaces.aspx">C# Interface Based Development</a>.</li>
+  <li>C# Corner, <a href="https://www.c-sharpcorner.com/article/C-Sharp-interface-based-development/">C# Interface Based Development</a>.</li>
```
Comment (jsoref) [inline csharp/ql/src/Bad Practices/Implementation Hiding/AbstractToConcreteCollection.qhelp:21]: This is just me happily following a redirect which coincidentally fixes the typo check-spelling complained about. As a bonus, it changes an `http` url to an `https` url.

File: go/ql/test/experimental/CWE-942/CorsMisconfiguration.go [hunk 6]
```diff
@@ -120,7 +120,7 @@ func main() {
 			}
 		})
 		http.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
-			// OK-ish: the input origin header is validated agains a whitelist.
+			// OK-ish: the input origin header is validated against a whitelist.
```
Comment (jsoref) [inline go/ql/test/experimental/CWE-942/CorsMisconfiguration.go:123]: I've mostly skipped `/test/` directories, but when I first started, this item jumped out at me, so it's thrown in at no extra charge.

File: go/ql/test/query-tests/Security/CWE-918/websocket.go [hunk 7]
```diff
@@ -96,7 +96,7 @@ func test() {
 	http.HandleFunc("/ex5", func(w http.ResponseWriter, r *http.Request) {
 		untrustedInput := r.Referer()
 
-		// good as input is tested againt regex
+		// good as input is tested against regex
```
Comment (jsoref) [inline go/ql/test/query-tests/Security/CWE-918/websocket.go:99]: Similarly, but a different spelling. And I don't think I'm actively ignoring `-tests`...

File: java/ql/src/Security/CWE/CWE-200/TempDirLocalInformationDisclosure.qhelp [hunk 8]
```diff
@@ -43,7 +43,7 @@ For example: <code>PosixFilePermissions.asFileAttribute(EnumSet.of(PosixFilePerm
 </example>
 
 <references>
-    <li>OSWAP: <a href="https://owasp.org/www-community/vulnerabilities/Insecure_Temporary_File">Insecure Temporary File</a>.</li>
+    <li>OWASP: <a href="https://owasp.org/www-community/vulnerabilities/Insecure_Temporary_File">Insecure Temporary File</a>.</li>
```
Comment (jsoref) [inline java/ql/src/Security/CWE/CWE-200/TempDirLocalInformationDisclosure.qhelp:46]: Brand

File: java/ql/src/Security/CWE/CWE-200/TempDirLocalInformationDisclosure.qhelp [hunk 9]
```diff
@@ -43,7 +43,7 @@ For example: <code>PosixFilePermissions.asFileAttribute(EnumSet.of(PosixFilePerm
...
     <li>CERT: <a href="https://wiki.sei.cmu.edu/confluence/display/java/FIO00-J.+Do+not+operate+on+files+in+shared+directories">FIO00-J. Do not operate on files in shared directories</a>.</li>
 </references>
 </qhelp>
```
Comment (jsoref) [inline java/ql/src/Security/CWE/CWE-200/TempDirLocalInformationDisclosure.qhelp:49]: Neither Git, GitHub, nor myself are fans of missing newline-at-eof. But I try not to change such things in PRs like this.

File: java/ql/src/Security/CWE/CWE-925/ImproperIntentVerification.qhelp [hunk 10]
```diff
@@ -26,7 +26,7 @@ Otherwise, a third-party application could impersonate the system this way to ca
 
 <recommendation>
 <p>
-In the <code>onReceive</code> method of a <code>BroadcastReciever</code>, the action of the received Intent should be checked. The following code demonstrates this.
+In the <code>onReceive</code> method of a <code>BroadcastReceiver</code>, the action of the received Intent should be checked. The following code demonstrates this.
```
Comment (jsoref) [inline java/ql/src/Security/CWE/CWE-925/ImproperIntentVerification.qhelp:29]: This makes the document agree w/ line 9...

File: java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.qhelp [hunk 11]
```diff
@@ -45,7 +45,7 @@ For this reason an initialization with a <code>null</code> environment is also v
 
 <p>The second example shows how a JMX Server is initialized securely if the <code>RMIConnectorServer</code> class is used.</p>
 
-<sample src="CorrectRMIConnectorServerEnvironmentInitalisation.java" />
+<sample src="CorrectRMIConnectorServerEnvironmentInitialisation.java" />
```
Comment (jsoref) [inline java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.qhelp:48]: This file is renamed above

File: python/ql/lib/semmle/python/web/flask/General.qll [hunk 12]
```diff
@@ -8,7 +8,7 @@ deprecated ClassValue theFlaskClass() { result = Value::named("flask.Flask") }
 /** Gets the flask MethodView class */
 deprecated ClassValue theFlaskMethodViewClass() { result = Value::named("flask.views.MethodView") }
 
-deprecated ClassValue theFlaskReponseClass() { result = Value::named("flask.Response") }
+deprecated ClassValue theFlaskResponseClass() { result = Value::named("flask.Response") }
```
Comment (jsoref) [inline python/ql/lib/semmle/python/web/flask/General.qll:11]: The previous thing at least has a comment. I can't tell if this is deprecate because of the typo or just to show off deprecation...

File: python/ql/test/library-tests/frameworks/modeling-example/NaiveModel.expected [hunk 13]
```diff
@@ -30,7 +30,7 @@ subpaths
 #select
 | test.py:22:10:22:24 | ControlFlowNode for Attribute() | test.py:21:11:21:18 | ControlFlowNode for source() | test.py:22:10:22:24 | ControlFlowNode for Attribute() | test flow (naive): test_simple |
 | test.py:33:10:33:12 | ControlFlowNode for val | test.py:29:11:29:18 | ControlFlowNode for source() | test.py:33:10:33:12 | ControlFlowNode for val | test flow (naive): test_alias |
-| test.py:41:10:41:12 | ControlFlowNode for val | test.py:45:11:45:18 | ControlFlowNode for source() | test.py:41:10:41:12 | ControlFlowNode for val | test flow (naive): test_accross_functions |
+| test.py:41:10:41:12 | ControlFlowNode for val | test.py:45:11:45:18 | ControlFlowNode for source() | test.py:41:10:41:12 | ControlFlowNode for val | test flow (naive): test_across_functions |
```
Comment (jsoref) [inline python/ql/test/library-tests/frameworks/modeling-example/NaiveModel.expected:33]: I'm not particularly certain about this change... If it's a problem, I'll drop it.

File: python/ql/test/library-tests/frameworks/modeling-example/test.py [hunk 14]
```diff
@@ -41,7 +41,7 @@ def sink_func(arg):
     sink(val)
 
 
-def test_accross_functions():
+def test_across_functions():
```
Comment (jsoref) [inline python/ql/test/library-tests/frameworks/modeling-example/test.py:44]: All the pieces should be changed, so it seems reasonable to me

**Review by jsoref — COMMENTED** [review]
Some items were automatically suggested by Google Sheets (and many more could have been had I fed it the input sooner).

All fault mine.

**Comment by jsoref** [comment]
Oh, I should note that https://github.com/check-spelling/codeql/pull/1 has roughly what this looks like as sarif output. Sadly, I don't think you can see much w/o being a member of the repository. Which means I need to find someone to talk to about how this stuff works. I was asked about the possibility of changing check-spelling to use sarif as its primary output mechanism, but if most people can't see its reports, that'd make it fairly painful. I'm very new to github's sarif handling and could use a hand understanding it, so if someone has time to talk... I'm on GitHub Security Lab Slack...

File: cpp/ql/src/Likely Bugs/Likely Typos/IncorrectNotOperatorUsage.qhelp (hunk 4 above)
Comment (geoffw0) [reply cpp/ql/src/Likely Bugs/Likely Typos/IncorrectNotOperatorUsage.qhelp:12]: Thanks for digging into this.

File: cpp/ql/src/Best Practices/Unused Entities/UnusedStaticVariables.qhelp (hunk 2 above)
Comment (geoffw0) [reply cpp/ql/src/Best Practices/Unused Entities/UnusedStaticVariables.qhelp:14]: Yes, `.qhelp` files are used to generate HTML, for example you can see this mistake here: https://codeql.github.com/codeql-query-help/cpp/cpp-unused-static-variable/

**Review by geoffw0 — COMMENTED** [review]
CPP looks good to me, with one point to discuss.

**Comment by aschackmull** [comment]
Thanks for submitting this PR!
Generally looks good to me, although I suspect that the changes in `python/ql/lib/semmle/python/web/flask/General.qll` and `python/ql/lib/semmle/python/web/flask/Response.qll` ought to be reverted as it otherwise might be a breaking change (and since it's deprecated, it's bound to go away at some point anyway).

File: cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp [hunk 15]
```diff
@@ -20,7 +20,7 @@ update if the requirements change (for example, a protocol is updated).
 </li>
 </ol>
 
-<p>This rule finds magic strings for which there is no pre-existing named
+<p>This rule finds magic strings for which there is no preexisting named
```
[the remark this answers is not in the document]
Comment (aschackmull) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp:23]: If indeed both are correct, then I think I also slightly prefer 'pre-existing' for the same reason, but I'll gladly defer to someone with a stronger opinion.

[the remark this answers is not in the document]
Comment (d10c) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp:23]: > British English often prefers hyphenated compounds, such as anti-smoking, whereas American English discourages the use of hyphens in compounds where there is no compelling reason, so antismoking is much more common. [—Wikipedia](https://en.wikipedia.org/wiki/American_and_British_English_spelling_differences?wprov=sfti1)

I suppose if American English is our target dialect in our documentation [which we do, IIRC; citation needed], then it would mean prioritizing the hyphenless spelling, though I personally agree the hyphenated spelling is easier to parse.

[the remark this answers is not in the document]
Comment (atorralba) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp:23]: > citation needed

https://github.com/github/codeql/blob/main/docs/qldoc-style-guide.md#language-requirements

[the remark this answers is not in the document]
Comment (tausbn) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp:23]: Obviously we should be doing like they do in The New Yorker, and write "preëxisting"...

Jokes aside, while I agree that we should usually follow American English, I think the version with the hyphen is more readable, and I think readability trumps correctness in this case.

File: java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.qhelp [hunk 16]
```diff
@@ -6,7 +6,7 @@
 <overview>
 <p>For special use cases some applications may implement a custom service which handles JMX-RMI connections.</p>
 
-<p>When creating such a custom service, a developer should pass a certain environment configuration to the JMX-RMI server initalisation, 
+<p>When creating such a custom service, a developer should pass a certain environment configuration to the JMX-RMI server initialisation,
```
Comment (tausbn) [inline java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.qhelp:9]:
```suggestion
<p>When creating such a custom service, a developer should pass a certain environment configuration to the JMX-RMI server initialization, 
```
(Since our style guide mandates the use of US English.)

File: python/ql/lib/semmle/python/web/flask/General.qll (hunk 12 above)
Comment (tausbn) [reply python/ql/lib/semmle/python/web/flask/General.qll:11]: This one is deprecated for other reasons, and I think it would be better to _not_ fix the typo here (as we would have to keep the old version around for backwards compatibility anyway).

File: python/ql/test/library-tests/frameworks/modeling-example/NaiveModel.expected (hunk 13 above)
Comment (tausbn) [reply python/ql/test/library-tests/frameworks/modeling-example/NaiveModel.expected:33]: Looks good to me. As long as both the test input and output reflect the typo fix, then everything should continue to work just fine.

**Review by tausbn — CHANGES_REQUESTED** [review]
Thanks for this! I have made a few comments and suggestions.

**Review by smowton — APPROVED** [review]
Go LGTM

**Review by hvitved — APPROVED** [review]
C# 👍 Thanks

File: cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp (hunk 15 above)
[the remark this answers is not in the document]
Comment (MathiasVP) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp:23]: Yeah, I think I'd also prefer the hyphenated version.

**Review by aibaars — APPROVED** [review]
:+1:  For the Ruby changes. Thanks!

**Comment by aibaars** [comment]
Note: we just merged https://github.com/github/codeql/pull/10753 to fix a bug in the `Render QHelp changes` CI job.

File: java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.qhelp (hunk 16 above)
Comment (jsoref) [reply java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.qhelp:9]: I'm willing to make this change (I personally favor en-US). Do note this repository has quite a few en-GB entries (including a number of other instances of `initialisation`). It'd probably be best, in _general_, if such corrections were done as a distinct PR (as is, this PR is intentionally scoped to just target `.qhelp` files).

I should really write a thing that makes it easy for me to flag en-GB or en-US entries -- it's something I haven't looked into in a while (mostly, I've decided to be skittish about this).

I'm going to toss in 3 related changes as a demo, and I'll tag them below. I can of course drop them / split them into another PR.

File: cpp/config/suites/cpp/metrics [hunk 17]
```diff
@@ -35,7 +35,7 @@
 + odasa-cpp-metrics/Classes/CInheritanceDepth.ql: /Metrics/Encapsulation
 + odasa-cpp-metrics/Classes/CLackOfCohesionCK.ql: /Metrics/Encapsulation
 + odasa-cpp-metrics/Classes/CSizeOfAPI.ql: /Metrics/Encapsulation
-+ odasa-cpp-metrics/Classes/CSpecialisation.ql: /Metrics/Encapsulation
++ odasa-cpp-metrics/Classes/CSpecialization.ql: /Metrics/Encapsulation
```
Comment (jsoref) [inline cpp/config/suites/cpp/metrics:38]: I'm not quite sure what `odasa-cpp-metrics` is (there are 10 files that mention it, but there are no actual file paths matching it)...

Anyway, this is/was `specialisation`

File: cpp/ql/src/jsf/4.07 Header Files/AV Rule 35.qhelp [hunk 18]
```diff
@@ -5,12 +5,12 @@
...
 <p>
 Some header files, such as those which define structures or classes, cannot be included more than once within a translation unit, as doing so would
-cause a redefinition error. Such headers must be guarded to prevent ill-effects from multiple inclusion. Simlarly, if header files include other
+cause a redefinition error. Such headers must be guarded to prevent ill-effects from multiple inclusion. Similarly, if header files include other
 header files, and this inclusion graph contains a cycle, then at least one file within the cycle must contain header guards in order to break the
 cycle. Because of cases like these, all headers should be guarded as a matter of good practice, even if they do not strictly need to be.
 </p>
 <p>
-Furthermore, most modern compilers contain optimisations which are triggered by header guards. If the header guard strictly conforms to the pattern
+Furthermore, most modern compilers contain optimizations which are triggered by header guards. If the header guard strictly conforms to the pattern
```
Comment (jsoref) [inline cpp/ql/src/jsf/4.07 Header Files/AV Rule 35.qhelp:13]: This is `optimisation`

**Comment by jsoref** [comment]
So, there are a couple of failures so far...

https://github.com/github/codeql/actions/runs/3218116407/jobs/5261890761

```
Run COMMENT_PREFIX="QHelp previews"
grep: comment_id.txt: No such file or directory
gh: Validation Failed (HTTP 422)
{"message":"Validation Failed","errors":[{"resource":"IssueComment","code":"unprocessable","field":"data","message":"Body is too long (maximum is 65536 characters)"}],"documentation_url":"https://docs.github.com/rest/reference/issues#create-an-issue-comment"}
Error: Process completed with exit code [1](https://github.com/github/codeql/actions/runs/3218116407/jobs/5261890761#step:4:1).
```

This feels like a bug in the CI -- I don't have the resources to address it here. (It's something the action I'm working on that I used here has tripped on, and for which [I have code to try to address](https://github.com/check-spelling/check-spelling/blob/f819b27a0f953841eb7ca0868a150ff0de9b212f/unknown-words.sh#L2093-L2141)...) 

https://github.com/github/codeql/actions/runs/3214105642/jobs/5261655998

```
A fatal error occurred: Failed to read path ./cpp/ql/src/Critical/aliasAnalysisWarning.inc.ql
(eventual cause: NoSuchFileException "./cpp/ql/src/Critical/aliasAnalysisWarning.inc.ql")
A fatal error occurred: Failed to read path ./cpp/ql/src/Critical/callGraphWarning.inc.ql
(eventual cause: NoSuchFileException "./cpp/ql/src/Critical/callGraphWarning.inc.ql")
A fatal error occurred: Failed to read path ./cpp/ql/src/Critical/dataFlowWarning.inc.ql
(eventual cause: NoSuchFileException "./cpp/ql/src/Critical/dataFlowWarning.inc.ql")
A fatal error occurred: Failed to read path ./cpp/ql/src/Critical/pointsToWarning.inc.ql
(eventual cause: NoSuchFileException "./cpp/ql/src/Critical/pointsToWarning.inc.ql")
A fatal error occurred: Failed to read path ./cpp/ql/src/Documentation/CommentedOutCodeQuery.inc.ql
(eventual cause: NoSuchFileException "./cpp/ql/src/Documentation/CommentedOutCodeQuery.inc.ql")
A fatal error occurred: Failed to read path ./cpp/ql/src/Likely Bugs/Leap Year/LeapYear.inc.ql
(eventual cause: NoSuchFileException "./cpp/ql/src/Likely Bugs/Leap Year/LeapYear.inc.ql")
Oops! A fatal internal error occurred.
com.semmle.util.exception.CatastrophicError: paragraph should be null
	at com.semmle.prettyprint.doc.QhelpToMD$QhelpSaxHandler.startParagraph(QhelpToMD.java:213)
	at com.semmle.prettyprint.doc.QhelpToMD$QhelpSaxHandler.startElement(QhelpToMD.java:449)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(Unknown Source)
```

This looks similar to https://github.com/github/codeql/pull/4953#issuecomment-760830987, but I don't have the resources to investigate.

https://github.com/github/codeql/actions/runs/3214105651/jobs/5261657025

```
Run EXIT_CODE=0
Previous HEAD position was ae[1](https://github.com/github/codeql/actions/runs/3214105651/jobs/5261657025#step:4:1)ceefe Merge d32a2b2aed31c87d2aeb7560d54493b83a8cc2e7 into 5cadd3c0e68334ef96c70d12e743f3489377beb5
HEAD is now at 5cadd3c0 Merge pull request #10712 from smowton/smowton/admin/kotlin-allow-single-embeddable-version
Warning: Missing QLdoc for classless-predicate General::theFlaskResponseClass/0
Error: Process completed with exit code 1.
```

?! -- This is interesting/surprising, because this looks like the code I specifically dropped per https://github.com/github/codeql/pull/10743#discussion_r991103334. Did this PR essentially poison the CI?

File: cpp/ql/lib/semmle/code/cpp/metrics/MetricClass.qll [hunk 19]
```diff
@@ -238,7 +238,7 @@ class MetricClass extends Class {
    * classes should add behavior to their base classes, but not alter that
    * behavior dramatically.
    */
-  float getSpecialisationIndex() {
+  float getSpecializationIndex() {
```
Comment (jketema) [inline cpp/ql/lib/semmle/code/cpp/metrics/MetricClass.qll:241]: Please revert this. There are special steps that need to be followed when renaming a predicate (aliasing the old name, marking that as deprecated, and writing up a change note). We might want to do that, but it's probably better done in a separate PR.

File: cpp/ql/src/Metrics/Classes/CSpecialization.ql [hunk 20]
```diff
@@ -3,7 +3,7 @@
  * @description The extent to which a subclass refines the behavior
  *              of its superclasses.
  * @kind treemap
- * @id cpp/specialisation-per-class
+ * @id cpp/specialization-per-class
```
Comment (jketema) [inline cpp/ql/src/Metrics/Classes/CSpecialization.ql:6]: This should probably be left as-is, as this will make if difficult for users to compare results produced by different versions of CodeQL.

File: cpp/ql/src/Metrics/Classes/CSpecialization.ql [hunk 21]
```diff
@@ -14,4 +14,4 @@ import cpp
 
 from Class c
 where c.fromSource()
-select c, c.getMetrics().getSpecialisationIndex()
+select c, c.getMetrics().getSpecializationIndex()
```
Comment (jketema) [inline cpp/ql/src/Metrics/Classes/CSpecialization.ql:17]: Please revert together with the predicate renaming.

File: csharp/ql/lib/semmle/code/csharp/Type.qll [hunk 22]
```diff
@@ -757,7 +757,7 @@ class RefType extends ValueOrRefType, @ref_type {
    *
    * This measures the inheritance complexity of this type.
    */
-  float getSpecialisationIndex() {
+  float getSpecializationIndex() {
```
Comment (jketema) [inline csharp/ql/lib/semmle/code/csharp/Type.qll:760]: Please revert. Same reasoning as before.

File: csharp/ql/src/Metrics/RefTypes/TSpecializationIndex.ql [hunk 23]
```diff
@@ -14,4 +14,4 @@ import csharp
 
 from RefType t
 where t.isSourceDeclaration()
-select t, t.getSpecialisationIndex() as n order by n desc
+select t, t.getSpecializationIndex() as n order by n desc
```
Comment (jketema) [inline csharp/ql/src/Metrics/RefTypes/TSpecializationIndex.ql:17]: Please revert. Same reasoning as before.

File: java/ql/lib/semmle/code/java/metrics/MetricRefType.qll [hunk 24]
```diff
@@ -290,7 +290,7 @@ class MetricRefType extends RefType, MetricElement {
    * should add behavior to their superclasses, but not alter that
    * behavior dramatically.
    */
-  float getSpecialisationIndex() {
+  float getSpecializationIndex() {
```
Comment (jketema) [inline java/ql/lib/semmle/code/java/metrics/MetricRefType.qll:293]: Please revert. Same reasoning as before.

File: java/ql/src/Metrics/RefTypes/TSpecializationIndex.ql [hunk 25]
```diff
@@ -17,4 +17,4 @@ where
   t.fromSource() and
   (t instanceof ParameterizedType implies t instanceof GenericType) and
   not t instanceof AnonymousClass
-select t, t.getMetrics().getSpecialisationIndex() as n order by n desc
+select t, t.getMetrics().getSpecializationIndex() as n order by n desc
```
Comment (jketema) [inline java/ql/src/Metrics/RefTypes/TSpecializationIndex.ql:20]: Please revert. Same reasoning as before.

**Review by jketema — CHANGES_REQUESTED** [review]
General comment: for file renames you will need to make sure that the references to those files are also updated. This is currently not the case see, e.g., `InitialisationNotRun.cpp` and the contents of `InitialisationNotRun.qlhelp`.

**Comment by jketema** [comment]
> So, there are a couple of failures so far...

You probably want to merge back the latest `main` per https://github.com/github/codeql/pull/10743#issuecomment-1273346064

It would probably also be a good idea to open a separate PR for any additional spelling mistakes you'd like to fix, as adding them here will make it more difficult to get this PR reviewed and merged.

File: java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.qhelp (hunk 16 above)
Comment (jsoref) [reply java/ql/src/experimental/Security/CWE/CWE-665/InsecureRmiJmxEnvironmentConfiguration.qhelp]: I've made this change to the file, including renaming the two referenced java files. I have not applied the change to `initialization` for the rest of the repository.

File: cpp/config/suites/cpp/metrics (hunk 17 above)
Comment (jsoref) [reply cpp/config/suites/cpp/metrics]: I've dropped this.

**Comment by aibaars** [comment]
> > > So, there are a couple of failures so far...
> 
> > You probably want to merge back the latest main per [#10743 (comment)](https://github.com/github/codeql/pull/10743#issuecomment-1273346064)
> 
> Oh. I missed that the PR was rebased. Maybe there are further bugs in the `Render QHelp changes` CI job? @aibaars

The job needed to be re-run, I just hit the `Approve and Run` button and QHelp seems to be fine now.

**Review by smowton — DISMISSED** [review]
Go still 👍

**Review by aschackmull — APPROVED** [review]

**Review by erik-krogh — DISMISSED** [review]
JS 👍

**Review by tausbn — DISMISSED** [review]
Python 👍

File: cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp (hunk 15 above)
[the remark this answers is not in the document]
Comment (jketema) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp:23]: I think we agree we want to keep the hyphenation, so please drop those changes.

[the remark this answers is not in the document]
Comment (jsoref) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp]: Dropped: https://github.com/github/codeql/compare/a931c1553d9cb079c55b8b4ff48d3487b693ece7..c92ce69f480f0b397fe6c1f2ed5ae934815b8398

[the remark this answers is not in the document]
Comment (jketema) [reply cpp/ql/src/Best Practices/Magic Constants/MagicConstantsString.qhelp]: Thanks.

**Review by aschackmull — APPROVED** [review]

**Review by geoffw0 — APPROVED** [review]
C++ LGTM.

**Review by jketema — APPROVED** [review]

**Comment by jketema** [comment]
This is now a subset of what was approved by everyone and all tests pass, so I'll go ahead and merge this.

**Comment by jketema** [comment]
@jsoref thanks a lot for the fixes!
